src/control/system/membership.go

```diff
@@ -711,6 +711,11 @@ func (m *Membership) handleEngineFailure(evt *events.RASEvent) {
 		return
 	}
 
+	if evt.Incarnation != 0 && evt.Incarnation < member.Incarnation {
```

Comment:
when would incarnation be zero and why should we skip this clause if it is?

Reply:
Fixed with the latest commit. Thanks!